miraclelinux/meta-debian-extended
# heimdal: Update from 7.5.0+dfsg-3+deb10u1 to 7.5.0+dfsg-3+deb10u2

# Purpose of pull request
This version fixes following CVEs.
- CVE-2022-45142

DLA: DLA 3311-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image 
2. Run qemu image
3. Run commands
4. Check libraries are installed correctly


### Build qemuarm64 image 
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " heimdal heimdal-clients heimdal-servers heimdal-kdc heimdal-kcm heimdal-dev heimdal-multidev heimdal-doc"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```

### Run commands
Run commands with --version to check they are at least works.
```
bin="
  afslog
  asn1_compile
  asn1_print
  gsstool
  heimtools
  hxtool
  kadmin.heimdal
  kdestroy.heimdal
  kf
  kgetcred
  kinit.heimdal
  klist.heimdal
  kpagsh kpasswd.heimdal
  krb5-config
  krb5-config.heimdal
  ksu.heimdal
  kswitch.heimdal
  ktutil.heimdal
  otp
  otpprint
  slc
  string2key
  verify_krb5_conf
"
for cmd in $bin; do
  /usr/bin/$cmd --version 2>1 >/dev/null ; echo $?;
done
```

### Check libraries
Check libraries are installed correctly.
```
libs="
  /usr/lib/libgssapi.so.3.0.0
  /usr/lib/libkrb5.so.26.0.0
  /usr/lib/libheimbase.so.1.0.0
  /usr/lib/heimdal/pkgconfig/krb5-gssapi.pc
  /usr/lib/heimdal/pkgconfig/krb5.pc
  /usr/lib/heimdal/pkgconfig/kadm-client.pc
  /usr/lib/heimdal/pkgconfig/kadm-server.pc
  /usr/lib/libasn1.so.8.0.0
  /usr/lib/heimdal-servers/kadmind
  /usr/lib/heimdal-servers/kdc
  /usr/lib/heimdal-servers/kfd
  /usr/lib/heimdal-servers/kpasswdd
  /usr/lib/libsl.so.0.2.1
  /usr/lib/libroken.so.18.1.0
  /usr/lib/libotp.so.0.1.5
  /usr/lib/libkadm5srv.so.8.0.1
  /usr/lib/libkdc.so.2.0.0
  /usr/lib/libkafs.so.0.5.1
  /usr/lib/libhdb.so.9.2.0
  /usr/lib/pkgconfig/heimdal-gssapi.pc
  /usr/lib/pkgconfig/heimdal-kadm-server.pc
  /usr/lib/pkgconfig/heimdal-kadm-client.pc
  /usr/lib/pkgconfig/krb5-gssapi.pc
  /usr/lib/pkgconfig/krb5.pc
  /usr/lib/pkgconfig/heimdal-krb5.pc
  /usr/lib/pkgconfig/kadm-client.pc
  /usr/lib/pkgconfig/kadm-server.pc
  /usr/lib/pkgconfig/kafs.pc
  /usr/lib/libkadm5clnt.so.7.0.1
  /usr/lib/libheimedit.so.0.0.36
  /usr/lib/libheimntlm.so.0.1.0
  /usr/lib/libcom_err.so.1.1.3
  /usr/lib/libhcrypto.so.4.1.0
  /usr/lib/libhx509.so.5.0.0
  /usr/lib/libheimsqlite.so.0.0.0
  /usr/lib/libwind.so.0.0.0
"
for lib in $libs; do ls $lib; done
```

## Test result
### Build package
Build succeeded.
```
$ bitbake heimdal
Loading cache: 100% |########################################################################################################################################| Time: 0:00:02
Loaded 2368 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |######################################################################################################################################| Time: 0:00:44
Parsing of 1351 .bb files complete (842 cached, 509 parsed). 2368 targets, 80 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.6"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:8f17d6cf6448ceaac5c7f35c538b91d477f79424"
meta-debian-extended = "update-heimdal_7.5.0+dfsg-3+deb10u2:a97f0e9ff56e2ac5aaf9b96a2f25c97894ff898f"
meta-emlinux         = "HEAD:ead20ef617c1af2d70f1e26e27a99ebd10160c80"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###################################################################################################################################| Time: 0:00:09
Sstate summary: Wanted 18 Found 11 Missed 7 Current 417 (61% match, 98% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2005 tasks of which 1979 didn't need to be rerun and all succeeded.
```
### Run commands
The commands is working at least.
```
root@qemuarm64:~# bin="
>   afslog
>   asn1_compile
>   asn1_print
>   gsstool
>   heimtools
>   hxtool
>   kadmin.heimdal
>   kdestroy.heimdal
>   kf
>   kgetcred
>   kinit.heimdal
>   klist.heimdal
>   kpagsh kpasswd.heimdal
>   krb5-config
>   krb5-config.heimdal
>   ksu.heimdal
>   kswitch.heimdal
>   ktutil.heimdal
>   otp
>   otpprint
>   slc
>   string2key
>   verify_krb5_conf
> "
root@qemuarm64:~# for cmd in $bin; do
>   /usr/bin/$cmd --version 2>1 >/dev/null ; echo $?;
> done
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
```
### Check libraries
The libraries is installed correctly.
```
root@qemuarm64:~# libs="
>   /usr/lib/libgssapi.so.3.0.0
>   /usr/lib/libkrb5.so.26.0.0
>   /usr/lib/libheimbase.so.1.0.0
>   /usr/lib/heimdal/pkgconfig/krb5-gssapi.pc
>   /usr/lib/heimdal/pkgconfig/krb5.pc
>   /usr/lib/heimdal/pkgconfig/kadm-client.pc
>   /usr/lib/heimdal/pkgconfig/kadm-server.pc
>   /usr/lib/libasn1.so.8.0.0
>   /usr/lib/heimdal-servers/kadmind
>   /usr/lib/heimdal-servers/kdc
>   /usr/lib/heimdal-servers/kfd
>   /usr/lib/heimdal-servers/kpasswdd
>   /usr/lib/libsl.so.0.2.1
>   /usr/lib/libroken.so.18.1.0
>   /usr/lib/libotp.so.0.1.5
>   /usr/lib/libkadm5srv.so.8.0.1
>   /usr/lib/libkdc.so.2.0.0
>   /usr/lib/libkafs.so.0.5.1
>   /usr/lib/libhdb.so.9.2.0
>   /usr/lib/pkgconfig/heimdal-gssapi.pc
>   /usr/lib/pkgconfig/heimdal-kadm-server.pc
>   /usr/lib/pkgconfig/heimdal-kadm-client.pc
>   /usr/lib/pkgconfig/krb5-gssapi.pc
>   /usr/lib/pkgconfig/krb5.pc
>   /usr/lib/pkgconfig/heimdal-krb5.pc
>   /usr/lib/pkgconfig/kadm-client.pc
>   /usr/lib/pkgconfig/kadm-server.pc
>   /usr/lib/pkgconfig/kafs.pc
>   /usr/lib/libkadm5clnt.so.7.0.1
>   /usr/lib/libheimedit.so.0.0.36
>   /usr/lib/libheimntlm.so.0.1.0
>   /usr/lib/libcom_err.so.1.1.3
>   /usr/lib/libhcrypto.so.4.1.0
>   /usr/lib/libhx509.so.5.0.0
>   /usr/lib/libheimsqlite.so.0.0.0
>   /usr/lib/libwind.so.0.0.0
> "
root@qemuarm64:~# for lib in $libs; do ls $lib ; done             
/usr/lib/libgssapi.so.3.0.0
/usr/lib/libkrb5.so.26.0.0
/usr/lib/libheimbase.so.1.0.0
/usr/lib/heimdal/pkgconfig/krb5-gssapi.pc
/usr/lib/heimdal/pkgconfig/krb5.pc
/usr/lib/heimdal/pkgconfig/kadm-client.pc
/usr/lib/heimdal/pkgconfig/kadm-server.pc
/usr/lib/libasn1.so.8.0.0
/usr/lib/heimdal-servers/kadmind
/usr/lib/heimdal-servers/kdc
/usr/lib/heimdal-servers/kfd
/usr/lib/heimdal-servers/kpasswdd
/usr/lib/libsl.so.0.2.1
/usr/lib/libroken.so.18.1.0
/usr/lib/libotp.so.0.1.5
/usr/lib/libkadm5srv.so.8.0.1
/usr/lib/libkdc.so.2.0.0
/usr/lib/libkafs.so.0.5.1
/usr/lib/libhdb.so.9.2.0
/usr/lib/pkgconfig/heimdal-gssapi.pc
/usr/lib/pkgconfig/heimdal-kadm-server.pc
/usr/lib/pkgconfig/heimdal-kadm-client.pc
/usr/lib/pkgconfig/krb5-gssapi.pc
/usr/lib/pkgconfig/krb5.pc
/usr/lib/pkgconfig/heimdal-krb5.pc
/usr/lib/pkgconfig/kadm-client.pc
/usr/lib/pkgconfig/kadm-server.pc
/usr/lib/pkgconfig/kafs.pc
/usr/lib/libkadm5clnt.so.7.0.1
/usr/lib/libheimedit.so.0.0.36
/usr/lib/libheimntlm.so.0.1.0
/usr/lib/libcom_err.so.1.1.3
/usr/lib/libhcrypto.so.4.1.0
/usr/lib/libhx509.so.5.0.0
/usr/lib/libheimsqlite.so.0.0.0
/usr/lib/libwind.so.0.0.0
```
